routes/api.js

```diff
@@ -1,10 +1,85 @@
 const express = require('express');
+const jwt    = require('jsonwebtoken');
+const config = require('../config/settings');
 const router = express.Router();
+const models = require('../database/models');
+
+const localUser = models.user_account;
+const progress = models.progress;
+
+router.use(function(req, res, next) {
+
+    // check header or url parameters or post parameters for token
+    let token;
+
+    if(!req.body) {
+        token = req.body.token;
+    } else if(!req.query) {
+        token = req.query.token;
+    } else {
+        token = req.headers['x-access-token'];
+    }
+
+    // decode token
+    if(token) {
+        // Removing quotes in the token
+        token = token.substring(1, token.length-1);
+
+        // verifies secret and checks exp
+        jwt.verify(token, config.secretKey, function(err, decoded) {
+            if(err) {
+                return res.json({ success: false, message: 'Failed to authenticate token.' });
+            } else {
+                // if everything is good, save to request for use in other routes
+                req.decoded = decoded;
+                next();
+            }
+        });
+
+    } else {
+        // if there is no token
+        // return an error
+        return res.status(403).send({
+            success: false,
+            message: 'No token provided.'
+        });
+    }
+});
 
 /* GET api listing. */
 router.get('/getUserInfo', (req, res) => {
     res.status(200).json({user: req.user});
 });
 
+router.get('/getProgressStatus', (req, res) => {
+    if(!req.user.email) {
```

Comment:
Yes, if the user is logged in then req.user is always defined. The user cannot access this api if he is not logged in.